src/Servers/IIS/IIS/test/Common.FunctionalTests/Infrastructure/EventLogHelpers.cs

```diff
@@ -65,12 +59,27 @@ public static string OnlyOneAppPerAppPool()
         }
     }
 
-    private static EventLogEntry[] AssertEntry(string regexString, IEnumerable<EventLogEntry> entries, bool allowMultiple = false)
+    private static async Task<EventLogEntry[]> AssertEntry(string regexString, IISDeploymentResult deploymentResult, bool allowMultiple = false)
     {
-        var expectedRegex = new Regex(regexString, RegexOptions.Singleline);
-        var matchedEntries = entries.Where(entry => expectedRegex.IsMatch(entry.Message)).ToArray();
-        Assert.True(matchedEntries.Length > 0, $"No entries matched by '{regexString}'");
-        Assert.True(allowMultiple || matchedEntries.Length < 2, $"Multiple entries matched by '{regexString}': {FormatEntries(matchedEntries)}");
+        // EventLogs don't seem to be instant, add retries to attempt to reduce test failures when looking for logs.
+        var retryCount = 10;
+        EventLogEntry[] matchedEntries = [];
+        while (retryCount > 0)
+        {
+            retryCount--;
+            var entries = GetEntries(deploymentResult);
+            var expectedRegex = new Regex(regexString, RegexOptions.Singleline);
+            matchedEntries = entries.Where(entry => expectedRegex.IsMatch(entry.Message)).ToArray();
+            if (matchedEntries.Length == 0 && retryCount > 0)
+            {
+                await Task.Delay(100);
+                continue;
+            }
+            Assert.True(matchedEntries.Length > 0, $"No entries matched by '{regexString}'");
+            Assert.True(allowMultiple || matchedEntries.Length < 2, $"Multiple entries matched by '{regexString}': {FormatEntries(matchedEntries)}");
+            return matchedEntries;
+        }
+
         return matchedEntries;
```

Comment:
We don't want to fail here?

Reply:
The way the loop is written it should never get here. So I guess it should throw an exception